WIP: Dbus

klali opened the pull request:
This is mostly to let anyone interested know that this quite big change is coming.

This is a partial re-write to not use bluepy but rather use the bluez dbus api directly.

klali committed:
convert the component to use bluez over dbus

klali committed:
enable periodic pings

klali committed:
give better errors if no bluetooth adapter is found

klali committed:
be consistent about plejd vs PLEJD

klali committed:
error handling etc

mudape commented:
Hi,
Running this on HassOS with Hass.io and get:

2020-01-04 21:28:39 ERROR (MainThread) [homeassistant.core] Error doing job: Future exception was never retrieved
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/ha-plejd-dbus/light.py", line 466, in _start_plejd
    connect(plejdinfo)
  File "/config/custom_components/ha-plejd-dbus/light.py", line 139, in connect
    import dbus
ModuleNotFoundError: **No module named 'dbus'**

klali commented:
Yes.. documentation is lagging behind. If you let homeassistant install the python modules you need (on debian/ubuntu) the packages libdbus-1-dev, libcairo2-dev & libgirepository1.0-dev if you install python packages manually you need dbus-python & PyGObject

I actually think it'd be better to use a simpler dbus module for this but the handiest examples I found where using dbus-python.

klali committed:
rework dbus support to use dbus-next which is pure python

this has the added benefit of being asyncio based, so this converts the
component to async as well

klali committed:
be consistent with capitalisation in logging

klali commented:
This branch now uses a different dbus library (dbus-next) which is pure python and should be much simpler to install. If you're interested in trying again and giving feedback that'd be welcome.

klali committed:
update documentation for the dbus based install

klali committed:
make discovery timeout configurable

klali committed:
run connect() straight from setup so we can fail with PlatformNotReady

this gets free retries

klali committed:
more debug log

klali committed:
mark update_state() as @callback and use async_schedule...()

klali committed:
logging fixes
